core/state_transition.go

```diff
@@ -513,7 +529,7 @@ func (st *StateTransition) refundGas(refundQuotient uint64) uint64 {
 	// Return ETH for remaining gas, exchanged at the original rate.
 	remaining := uint256.NewInt(st.gasRemaining)
 	remaining.Mul(remaining, uint256.MustFromBig(st.msg.GasPrice))
-	st.state.AddBalance(st.msg.From, remaining, tracing.BalanceIncreaseGasReturn)
+	st.state.AddBalance(*st.msg.GasPayer, remaining, tracing.BalanceIncreaseGasReturn)
```

Comment:
This change makes no sense to me. If `IsRip7560Frame` is refunded separately, the `GasPayer` here will always be a `From` address, right?